src/Workspaces/Core/Portable/Rename/RenameLocation.ReferenceProcessing.cs

```diff
@@ -337,12 +344,20 @@ private static string TrimNameToAfterLastDot(string name)
                             {
                                 if (location.IsInSource)
                                 {
-                                    var token = location.FindToken(cancellationToken);
-                                    if (!syntaxFacts.IsReservedOrContextualKeyword(token) &&
-                                        token.ValueText == referencedSymbol.Name)
+                                    var document = solution.GetRequiredDocument(location.SourceTree);
+
+                                    // If the location is in a source generated file, we won't rename it. Our assumption in this case is we
+                                    // have cascaded to this symbol from our original source symbol, and the generator will update this file
+                                    // based on the renamed symbol.
+                                    if (document is not SourceGeneratedDocument)
                                     {
-                                        Contract.ThrowIfNull(location.SourceTree);
-                                        results.Add(new RenameLocation(location, solution.GetRequiredDocument(location.SourceTree).Id));
+                                        var token = location.FindToken(cancellationToken);
+                                        if (!syntaxFacts.IsReservedOrContextualKeyword(token) &&
+                                            token.ValueText == referencedSymbol.Name)
+                                        {
+                                            Contract.ThrowIfNull(location.SourceTree);
+                                            results.Add(new RenameLocation(location, document.Id));
```

Comment:
would it possibly be useful to have a static local to add items to the list that first checks if the document is not source generated (so you don't have to copy the comment)

Reply:
Good idea.

Reply:
Updated, but didn't make it static; it's handy because if it's non-static it then can also grab the solution and the output array so it's nice and tight.